src/spotify_webapi.c

```diff
@@ -240,13 +240,24 @@ tokens_get(struct keyval *kv, const char **err)
       goto out_free_input_body;
     }
 
+  if (spotify_access_token)
+    {
+      free(spotify_access_token);
+      spotify_access_token = NULL;
+    }
+
   tmp = jparse_str_from_obj(haystack, "access_token");
   if (tmp)
-    spotify_access_token  = strdup(tmp);
+    spotify_access_token = strdup(tmp);
 
   tmp = jparse_str_from_obj(haystack, "refresh_token");
   if (tmp)
-    spotify_refresh_token = strdup(tmp);
+    {
+      if (spotify_refresh_token)
```

Comment:
Learnt from the other thread that free(null) is a no-op, so the if can be removed.